Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/consent/SubmissionConsentViewModel.kt

```diff
@@ -14,13 +15,14 @@ import de.rki.coronawarnapp.ui.submission.viewmodel.SubmissionNavigationEvents
 import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
 import de.rki.coronawarnapp.util.ui.SingleLiveEvent
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
-import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
 import kotlinx.coroutines.flow.first
 import timber.log.Timber
 
 class SubmissionConsentViewModel @AssistedInject constructor(
     interoperabilityRepository: InteroperabilityRepository,
     dispatcherProvider: DispatcherProvider,
+    @Assisted private val qrCode: String,
```

Comment:
Should not this be `CoronaTestQRCode` ? which is already parsed by the scanner

Reply:
Not necessarily. We can use the `rawString` from the scan result. And the consent fragment also had this `qrCode` attribute that was being passed from `MainActivity`.

Reply:
this is what get from the scanner , why you need to validate it again ?